mail/protocols/imap/src/main/java/com/fsck/k9/mail/store/imap/RealImapFolder.kt

```diff
@@ -288,19 +272,6 @@ internal class RealImapFolder(
         val encodedDestinationFolderName = folderNameCodec.encode(folder.prefixedName)
         val escapedDestinationFolderName = ImapUtility.encodeString(encodedDestinationFolderName)
 
-        // TODO: Just perform the operation and only check for existence of the folder if the operation fails.
-        if (!exists(escapedDestinationFolderName)) {
-            if (K9MailLib.isDebug()) {
-                Timber.i(
-                    "ImapFolder.copyMessages: couldn't find remote folder '%s' for %s",
-                    escapedDestinationFolderName,
-                    logId,
-                )
-            }
-
-            throw FolderNotFoundException(folder.serverId)
```

Comment:
We're not actually using this exception anywhere. So getting rid of this code doesn't change the behavior.